core/server/api/shared/http.js

```diff
@@ -27,6 +27,10 @@ const http = (apiImpl) => {
 
         apiImpl(frame)
             .then((result) => {
+                return shared.headers.get(result, apiImpl.headers, frame)
+                    .then(headers => ({result, headers}));
```

Comment:
This is an ugly bit imo. @kirrg001 maybe you have a better idea on a more elegant syntax for passing multiple objects further into `.then` chain?